web.config

```diff
@@ -10,36 +10,130 @@
     <system.webServer>
         <httpProtocol>
             <customHeaders>
+                <!-- Remove unnecesary headers -->
                 <remove name="X-Powered-By"/>
                 <remove name="server"/>
-                <add name="x-dns-prefetch-control" value="on"/>
+                <!-- Security headers -->
                 <add name="X-Content-Type-Options" value="nosniff"/>
                 <add name="Strict-Transport-Security" value="max-age=31536000"/>
             </customHeaders>
         </httpProtocol>
         <!-- Visit http://blogs.msdn.com/b/windowsazure/archive/2013/11/14/introduction-to-websockets-on-windows-azure-web-sites.aspx for more information on WebSocket support -->
         <webSocket enabled="false"/>
-        <handlers>
+        <handlers> -->
             <!-- Indicates that the app.js file is a node.js site to be handled by the iisnode module -->
             <add name="iisnode" path="src/server/index.js" verb="*" modules="iisnode"/>
         </handlers>
+
+        <!--
+            We still have some dynamic parts on the site that we want to compress. E.g.:
+             * JSON responses from the server
+             * dynamically generated html
+             * html pages served via `/` and not ending on `.html` (haven't figured out a way to do this mixed with our dynamic part)
+        -->
+        <urlCompression doStaticCompression="true" doDynamicCompression="true" dynamicCompressionBeforeCache="false" />
+        <staticContent>
+            <!-- We set the mimeType for all the types we are going to use in the site. IIS supports a few of those but
+                 they not always have the right values so we are explicit.
+                 Also we set `cache-control: no-cache` by default. We will override this based on the file's path
+                 See https://docs.microsoft.com/en-us/iis/configuration/system.webserver/staticcontent/clientcache for more info -->
+            <clientCache cacheControlMode="DisableCache" />
+            <!-- The brotli mime type is unknown to IIS -->
+            <remove fileExtension=".br" />
+            <mimeMap fileExtension=".br" mimeType="application/brotli" />
```

Comment:
Where did you get `application/brotli` from? Does IIS need this?

Reply:
See also: https://groups.google.com/forum/#!topic/brotli/ie76SizwcEE.

Reply:
TBH, this is what I had in the `web.config` of another site. By default IIS doesn't serve any image if it doesn't have a mime type so it could be because of that. I'll remove it and see what happens.

Reply:
Removing it breaks everything so we need it.

Reply:
:(

Can you add a comment?